bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/kostalinverter/internal/secondgeneration/SecondGenerationHandler.java

```diff
@@ -0,0 +1,410 @@
+        this.inverterConfig = inverterConfig;
+
+        // Initialize Gson object gson
+        final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        this.gson = gson;
+
+        // Temporary value during initializing
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Start update as configured
+        secondGenerationPoller = scheduler.scheduleWithFixedDelay(() -> {
+            try {
+                refresh();
+                updateStatus(ThingStatus.ONLINE);
+            } catch (RuntimeException scheduleWithFixedDelayException) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        scheduleWithFixedDelayException.getClass().getName() + ":"
+                                + scheduleWithFixedDelayException.getMessage());
+                logger.debug("Error when connecting to inverter, {}: {}", getThing().getUID(),
+                        scheduleWithFixedDelayException.getMessage());
+            } catch (InterruptedException e) {
+                logger.debug("Communication with inverter interrupted, exception {}", e.getMessage());
+            } catch (ExecutionException e) {
+                logger.debug("Communication with inverter failed, exception {}", e.getMessage());
+            } catch (TimeoutException e) {
+                logger.debug("Communication with inverter timed out, exception {}", e.getMessage());
+            }
+        }, 0, SecondGenerationInverterConfig.REFRESHINTERVAL_SEC, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        final ScheduledFuture<?> secondGenerationLocalPoller = secondGenerationPoller;
+
+        if (secondGenerationLocalPoller != null) {
+            secondGenerationLocalPoller.cancel(true);
+            secondGenerationPoller = null;
+        }
+    }
+
+    @SuppressWarnings({ "null", "unchecked" })
+    private void refresh() throws InterruptedException, ExecutionException, TimeoutException {
+        // dxsEntries part
+        String dxsEntriesCall = inverterConfig.url.toString() + "/api/dxs.json?dxsEntries="
+                + channelConfigs.get(0).dxsEntries;
+        for (int i = 1; i < channelConfigs.size(); i++) {
+            dxsEntriesCall += ("&dxsEntries=" + channelConfigs.get(i).dxsEntries);
+        }
+        String jsonDxsEntriesResponse = callURL(dxsEntriesCall);
+        SecondGenerationDxsEntriesContainerDTO dxsEntriesContainer = gson.fromJson(jsonDxsEntriesResponse,
+                SecondGenerationDxsEntriesContainerDTO.class);
+
+        String[] channelPosts = new String[23];
+        int channelPostsCounter = 0;
+        for (SecondGenerationDxsEntries dxsentries : dxsEntriesContainer.dxsEntries) {
+            channelPosts[channelPostsCounter] = dxsentries.getName();
+            channelPostsCounter++;
+        }
+        List<String> channelPostsTemp = new ArrayList<String>(Arrays.asList(channelPosts));
```

Comment:
Syntactical sugar. Same for the others.
```suggestion
        List<String> channelPostsTemp = List.of(channelPosts);
```